test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.server.cs

```diff
@@ -75,6 +75,7 @@ public void GrpcAspNetCoreInstrumentationAddsCorrectAttributes(bool? enableGrpcA
             var client = new Greeter.GreeterClient(channel);
             client.SayHello(new HelloRequest());
 
+            this.server.Dispose();
```

Comment:
Arrgh! Yea these tests have been finicky. One of the issues is our usage of `WaitForProcessorInvocations(processor, 2);`. If the method invocations on the processor changes, tests like this can become flaky.

For example, here's another similar test that waits for more invocations:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/41a0a266495888e1b6f147a188a9c23199221ccb/test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.client.cs#L72

This test is accounting for the fact that `SetParentProvider` is called. Instead of calling this.server.Dispose you could probably use `WaitForProcessorInvocations(processor, 3); // SetParentProvider/OnStart/OnEnd`

Since the tracerProvider does not get disposed until the end of the unit test you will not see an invocation to OnShutdown or Dispose on the TracerProvider.